internals/plan/plan.go

```diff
@@ -1125,7 +1319,21 @@ func ParseLayer(order int, label string, data []byte) (*Layer, error) {
 		return nil, err
 	}
 
-	return &layer, err
+	return layer, err
+}
+
+// mapHasKeys returns true if the key list supplied is an exact match of the
+// keys in the map (ordering is ignored).
+func mapHasKeys[M ~map[K]V, K comparable, V any](inMap M, keyList []K) bool {
+	if len(inMap) != len(keyList) {
+		return false
+	}
+	for _, key := range keyList {
+		if _, ok := inMap[key]; !ok {
```

Comment:
FWIW, this implementation is not fool-proof: if you have N times the exact same key in keyList, where N matches len(inMap), this function still returns true.

Reply:
Fixed, thank you! Resolving.